common/db.go

```diff
@@ -371,6 +381,23 @@ func (db *DB) SetStopReason(id *big.Int, reason string) error {
 	return nil
 }
 
+func (db *DB) ReceiptExists(jobID *big.Int, seqNo uint64) (bool, error) {
+	if db == nil {
+		return false, nil
+	}
+	glog.V(DEBUG).Infof("db: Checking receipt %v for job %v", seqNo, jobID)
+	var res int
+	err := db.checkRec.QueryRow(jobID.Int64(), seqNo).Scan(&res)
+	if err != nil && err == sql.ErrNoRows {
+		return false, nil
+	}
+	if err != nil {
+		glog.Error("Could not check receipt existence ", err)
+		return false, err // should we return true given the intended usage?
```

Comment:
Re: comment - seems ok to return false here since if the error is not `sql.ErrNoRows` and is instead something like `ErrConnDone` for a closed DB connection we don't actually know if the receipt exists or not

Reply:
Since the current usage of `ReceiptExists` is to bail out if a receipt actually exists, I wasn't sure if we wanted to err on the side of caution, in case we later call `CheckReceipt` somewhere else, and ignore the error code for some reason. But that's quite speculative on several levels. Removed the comment.

This is a really good example of where algebraic data types (or even just option types) make the semantics of a function much cleaner.